flink-runtime/src/main/java/org/apache/flink/runtime/zookeeper/ZooKeeperStateHandleStore.java

```diff
@@ -375,7 +415,7 @@ public IntegerResourceVersion exists(String pathInZooKeeper) throws Exception {
                     try {
                         final RetrievableStateHandle<T> stateHandle = getAndLock(path);
                         stateHandles.add(new Tuple2<>(stateHandle, path));
-                    } catch (KeeperException.NoNodeException ignored) {
+                    } catch (NotExistException ignored) {
                         // Concurrent deletion, retry
                         continue retry;
```

Comment:
oh man, goto statement spotted 🙈

Reply:
Yes, but I left it there because I didn't want to touch the code more than necessary considering that we might want to get rid of it, anyway, soon'ish

Reply:
👍 sure, this was an off-topic comment